src/components/figures/figure_chart/figure_chart.xml

```diff
@@ -20,7 +20,7 @@
       />
 
       <Menu
-        t-if="menuState.isOpen"
+        t-if="menuState.isOpen and !env.model.getters.isReadonly()"
```

Comment:
> You should prevent one from being able to open the menu (by which I mean trigger the change of state) instead of hiding the menu afterwards. With the current proposition, if I were to consult the internal state of the component, I would see that the menu is open but it would not reflect in the browser.

Thank you for pointing that out! I see your point. With the previous solution, In readonly mode if we right click on the chart and change mode to access, the context menu would become visible.